trl/trainer/dpo_trainer.py

```diff
@@ -1037,10 +1049,13 @@ def concatenated_inputs(
             )
 
         if is_vision_model:
-            concatenated_batch["pixel_values"] = batch["prompt_pixel_values"].repeat(2, 1, 1, 1, 1).to(device=device)
-            concatenated_batch["pixel_attention_mask"] = (
-                batch["prompt_pixel_attention_mask"].repeat(2, 1, 1, 1).to(device=device)
+            concatenated_batch["pixel_values"] = torch.cat(
+                [batch["prompt_pixel_values"], batch["prompt_pixel_values"]], dim=0
             )
+            if "prompt_pixel_attention_mask" in batch:
+                concatenated_batch["pixel_attention_mask"] = torch.cat(
+                    [batch["prompt_pixel_attention_mask"], batch["prompt_pixel_attention_mask"]], dim=0
+                )
```

Comment:
depending on the processor, the `prompt_pixel_values` can have a shape `(B, N, 3, W, H)` (idefics2) or `(B, 3, W, H)` (Llava). To make these lines compatible with both cases, we use self concatenation cat in the first dim instead of `repeat`.

Reply:
you mean `dim=1`?

Reply:
At this point, `pixel_values` is either 
- a batch of images (llava case), so shape `(B, C, W, H)`, or 
- a batch of list of images (idefics2 case), so shape `(B, N, C, W, H)`

The image is shared between chosen and rejected. So we need to duplicate the batch in the batch dimension. You want to end up with a shape  `(2*B, ...)`
- first case: `(2*B, C, W, H)`
- second case: `(2*B, N, C, W, H)`

The problem is that with `torch.repeat`, you need to pass the number of repetition in each dimension. So having something like

```python
if len(image.shape) == 4:
    pixel_values = prompt_pixel_values.repeat(2, 1, 1, 1)
elif len(image.shape) == 5:
    pixel_values  = prompt_pixel_values.repeat(2, 1, 1, 1, 1)
```

An easier solution is to write:

```python
pixel_values = torch.cat([prompt_pixel_values, prompt_pixel_values], dim=0)
```